context/src/test/java/io/opentelemetry/context/LazyStorageTest.java

```diff
@@ -76,6 +76,13 @@ public void set_storage_provider_property_matches_one_provides() throws Exceptio
     }
   }
 
+  @Test
+  public void enforce_default_thread_local_storage() {
```

Comment:
I don't think we can test this sort of behavior with other ones since the system property is only loaded once. Can you extract another testset for this case? I think `context` will be full of many test sets because of his global initialization logic :)

Reply:
Actually since we're not testing the public method it looks ok. But we should still make sure the system properties are cleaned up. I realize this already affects the existing tests, but can you use junit pioneer?

https://github.com/open-telemetry/opentelemetry-java/blob/71ba8e1b87bfd16f2ecd21643bf55b88d1ecfcaa/sdk/common/src/test/java/io/opentelemetry/sdk/common/export/ConfigBuilderTest.java#L238

Reply:
I already clearup the properties in aftereach. It is not enough?
```
@AfterEach
  public void after() {
    System.clearProperty(CONTEXT_STORAGE_PROVIDER_PROPERTY);
  }
```

Reply:
Using the annotation is more self-documenting and I guess less error-prone. It's not so clear which tests depend on that `AfterEach` existing, so the annotation is nice to make it very self-documenting / clear. It's not a blocking change but I think it's a good idea.

Reply:
You are right. Thanks for help. That makes it more simpler and clearer. : )